packages/@aws-cdk/aws-cloudfront/lib/web_distribution.ts

```diff
@@ -791,6 +807,8 @@ export class CloudFrontWebDistribution extends cdk.Construct implements IDistrib
             originSslProtocols: originConfig.customOriginSource.allowedOriginSSLVersions || [OriginSslPolicy.TLS_V1_2],
           }
           : undefined,
+        connectionAttempts: originConfig.connectionAttempts || 3,
```

Comment:
use `??`, especially for integer values. (if `0` was a legal value, this would still default to `3`)